### PR TITLE
[RHCLOUD-19197] feature: "provisioning" app type

### PR DESCRIPTION
**Pull request by MikelAlejoBR**
They are starting to integrate with us and they need this application
type.

## Links

[[RHCLOUD-19197]](https://issues.redhat.com/browse/RHCLOUD-19197)

### CODE REVIEW

File: dao/seeds/source_types.yml [hunk 1]
```diff
@@ -66,6 +66,24 @@ amazon:
...
+        - name: authentication.username
+          component: text-field
+          label: ARN
+          isRequired: true
+          validate:
+            - type: required
+            - type: pattern
+              pattern: "^arnaws:.*"
+            - type: min-length
+              threshold: 10
```
Comment (lindgrenj6) [inline dao/seeds/source_types.yml:86]: These just need moved 2 spaces to the left - fields is an array of objects.

Comment (MikelAlejoBR) [reply dao/seeds/source_types.yml:86]: I've reduced the indentation in two spaces. I was confused by this, though:

> fields is an array of objects.

I think that these are equivalent?

```yaml
array:
  - one
  - two
  - three
```

```yaml
array:
- one
- two
- three
```

Comment (lindgrenj6) [reply dao/seeds/source_types.yml:86]: They might be treated the same by some parsers but not by others - thats where it gets fun. For example parsing the second one in ruby works, but the first one fails. For the k8s Go library it works, and for gopkg.in/yaml v2 & v3 it works.


tl;dr: yaml sucks. but the whitespace usually denotes an object rather than an array.

Comment (MikelAlejoBR) [reply dao/seeds/source_types.yml:86]: Wow, TIL. [Yaml Spec V1.2 §10.1.1.2.](https://yaml.org/spec/1.2.2/#10112-generic-sequence)

**Review by petracihalova — APPROVED** [review]

**Review by lpichler — APPROVED** [review]

File: dao/seeds/application_types.yml [hunk 2]
```diff
@@ -69,3 +69,11 @@
   supported_authentication_types:
     satellite:
     - receptor_node
+"/insights/platform/provisioning-service":
+  display_name: Provisioning service
```
Comment (ezr-ondrej) [inline dao/seeds/application_types.yml]: I'd rather go with Provisioning only, but unimportat ATM :)
```suggestion
  display_name: Provisioning
```

File: dao/seeds/source_types.yml [hunk 3]
```diff
@@ -66,6 +66,24 @@ amazon:
           pattern: "^arnaws:.*"
         - type: min-length
           threshold: 10
+    - type: provisioning-service-arn
+      name: Provisioning service's ARN
```
Comment (lzap) [inline dao/seeds/source_types.yml]: Just not to get this skipped, it's also here:

```suggestion
      name: Provisioning's ARN
```

**Comment by MikelAlejoBR** [comment]
All right thank you for your inputs @ezr-ondrej and @lzap . I've modified the name. @lpichler, I think this is ready :)

**Review by lpichler — APPROVED** [review]
